storybook/src/components/SearchField/SearchField.test.stories.tsx

```diff
@@ -18,12 +19,28 @@ export default meta
 
 type Story = StoryObj<typeof meta>
 
+const testFn = fn()
+
 export const Test: Story = {
+  args: {
+    onSubmit: (event) => {
+      event.preventDefault()
+      const form = event.target as HTMLFormElement
+      const searchInput = form.querySelector('input[type="search"]') as HTMLInputElement
+      const searchValue = searchInput?.value
+      testFn(searchValue)
+    },
+  },
+  play: async ({ canvas, userEvent }) => {
+    await userEvent.type(canvas.getByTestId('searchbox'), 'Amsterdam')
+    await userEvent.click(canvas.getByTestId('button'))
+    expect(testFn).toHaveBeenCalledWith('Amsterdam')
+  },
   render: (args) => (
     <div style={{ display: 'flex', flexDirection: 'column', gap: 'var(--ams-space-s)' }}>
       <SearchField {...args}>
-        <SearchField.Input placeholder="Waar ben je naar opzoek?" />
-        <SearchField.Button />
+        <SearchField.Input data-testid="searchbox" placeholder="Waar ben je naar opzoek?" />
+        <SearchField.Button data-testid="button" onClick={testFn} />
```

Comment:
I don't think we need to define the test function as the `onClick` handler here. We only want to test the submitted values, right?

```suggestion
        <SearchField.Button data-testid="button" />
```